sky/tools/skyanalyzer

```diff
@@ -32,6 +32,9 @@ _IGNORED_PATTERNS = [
   re.compile(r'^\[hint\] .+ requires dynamic invoke'), # too many false positives, e.g. https://github.com/dart-lang/sdk/issues/24564
   re.compile(r'^\[hint\] Runtime check on non-ground type .+ may throw StrongModeError'), # https://github.com/dart-lang/sdk/issues/24565
   re.compile(r'^\[hint] \([_, ]+\) .+ has inferred type '), # ignore underscore-only arguments
+  re.compile(r'^\[hint] .+ has inferred type \(.+\) '), # ignore inferred function types
```

Comment:
I felt that treating these hints as errors was overkill.

List literals without type annotations are all over the codebase.  I started going down the path of adding annotations but decided that it wasn't worthwhile.  The hint indicates that the type of the list can be inferred - for example, from a function argument.  If the caller includes a list entry that doesn't match the argument's type, then skyanalyzer will flag that as an error.

The same applies to function types.  And afaik there is no way in Dart to declare an inline annotation of a function type.  You need to declare a separate typedef, which imho isn't improving readability.